WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListView.swift

```diff
@@ -63,6 +57,81 @@ final class BlazeCampaignListHostingController: UIHostingController<BlazeCampaig
     }
 }
 
+// MARK: - Navigation related
+private extension BlazeCampaignListHostingController {
+    /// Handles navigation to the different campaign creation views.
+    func navigateToCampaignCreation() {
+        switch viewModel.createCampaignDestination {
+        case .productSelector:
+            navigateToBlazeProductSelector(source: .campaignList)
+        case .campaignForm(let productID):
+            navigateToNativeCampaignCreation(source: .campaignList, productID: productID)
+        case .webviewForm(let productID):
+            navigateToWebCampaignCreation(source: .campaignList, productID: productID)
+        case .noProductAvailable:
+            break // TODO 11685: add error alert.
+        }
+    }
+
+    /// Handles navigation to the webview Blaze creation
+    func navigateToWebCampaignCreation(source: BlazeSource, productID: Int64? = nil) {
+        let webViewModel = BlazeWebViewModel(siteID: viewModel.siteID,
+                source: source,
+                siteURL: viewModel.siteURL,
+                productID: productID) { [weak self] in
+            self?.handlePostCreation()
+        }
+        let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+        self.navigationController?.show(webViewController, sender: self)
```

Comment:
`self` not needed here.

Reply:
This is removed completely as part of refactor, thanks!